commit 087b681805f1de084f89f1041af67295aa981192
Author: Daniele Ceraolo Spurio <daniele.ceraolospurio@intel.com>
Date:   Tue Sep 27 17:41:40 2022 -0700

    drm/i915/dg2: setup HuC loading via GSC
    
    The GSC will perform both the load and the authentication, so we just
    need to check the auth bit after the GSC has replied.
    Since we require the PXP module to load the HuC, the earliest we can
    trigger the load is during the pxp_bind operation.
    
    Note that GSC-loaded HuC survives GT reset, so we need to just mark it
    as ready when we re-init the GT HW.
    
    V2: move setting of HuC fw error state to the failure path of the HuC
    auth function, so it covers both the legacy and new auth flows
    V4:
    1. Fix typo in the commit message
    2. style fix in intel_huc_wait_for_auth_complete()
    
    Signed-off-by: Daniele Ceraolo Spurio <daniele.ceraolospurio@intel.com>
    Signed-off-by: Vitaly Lubart <vitaly.lubart@intel.com>
    Signed-off-by: Tomas Winkler <tomas.winkler@intel.com>
    Reviewed-by: Alan Previn <alan.previn.teres.alexis@intel.com>
    Link: https://patchwork.freedesktop.org/patch/msgid/20220928004145.745803-11-daniele.ceraolospurio@intel.com

diff --git a/drivers/gpu/drm/i915/gt/uc/intel_huc.h b/drivers/gpu/drm/i915/gt/uc/intel_huc.h
index d7e25b6e879e..51f9d96a3ca3 100644
--- a/drivers/gpu/drm/i915/gt/uc/intel_huc.h
+++ b/drivers/gpu/drm/i915/gt/uc/intel_huc.h
@@ -26,8 +26,10 @@ void intel_huc_init_early(struct intel_huc *huc);
 int intel_huc_init(struct intel_huc *huc);
 void intel_huc_fini(struct intel_huc *huc);
 int intel_huc_auth(struct intel_huc *huc);
+int intel_huc_wait_for_auth_complete(struct intel_huc *huc);
 int intel_huc_check_status(struct intel_huc *huc);
 void intel_huc_update_auth_status(struct intel_huc *huc);
+bool intel_huc_is_authenticated(struct intel_huc *huc);
 
 static inline int intel_huc_sanitize(struct intel_huc *huc)
 {